R/utils.R

```diff
@@ -78,15 +119,33 @@ find_phantom <- function() {
 #' writable, the directory \file{PhantomJS} under the installation directory of
 #' the \pkg{webshot} package will be tried. If this directory still fails, you
 #' will have to install PhantomJS by yourself.
+#'
+#' If PhantomJS is not already installed on the computer, this function will
+#' attempt to install it. However, if the version of PhantomJS installed is
+#' greater than or equal to the requested version, this function will not
+#' perform the installation procedure again unless the \code{force} parameter is
+#' set to \code{TRUE}. As a result, this function may also be used to reinstall or
+#' downgrade the version of PhantomJS found.
+#'
 #' @param version The version number of PhantomJS.
 #' @param baseURL The base URL for the location of PhantomJS binaries for
 #'   download. If the default download site is unavailable, you may specify an
 #'   alternative mirror, such as
 #'   \code{"https://bitbucket.org/ariya/phantomjs/downloads/"}.
+#' @param force Install PhantomJS even if the version installed is the
+#'   latest or if the requested version is older. This is useful to reinstall
+#'   or downgrade the version of PhantomJS.
 #' @return \code{NULL} (the executable is written to a system directory).
 #' @export
 install_phantomjs <- function(version = '2.1.1',
-    baseURL = 'https://github.com/wch/webshot/releases/download/v0.3.1/') {
+    baseURL = 'https://github.com/wch/webshot/releases/download/v0.3.1/',
+    force = FALSE) {
+
+  if (is_phantomjs_version_latest(version) && !force) {
+      message('It seems that the installed version of `phantomjs` is the latest. ',
+              'To reinstall the version presently installed, use `force = TRUE`.')
```

Comment:
Slight correction: this won't necessarily install the presently-installed version; it will install the version specified by the `version` parameter.

Reply:
I updated the message to reflect this.